doc/espresso-dev-node.md

```diff
@@ -0,0 +1,71 @@
+| `sequencer_api_port`            | `u16`           | `ESPRESSO_SEQUENCER_API_PORT`        | Required                                                      | Port that the HTTP API will use.                                                                                                             |
+| `sequencer_api_max_connections` | `Option<usize>` | `ESPRESSO_SEQUENCER_MAX_CONNECTIONS` | None                                                          | Maximum concurrent connections allowed by the HTTP API server.                                                                               |
+| `builder_port`                  | `Option<u16>`   | `ESPRESSO_BUILDER_PORT`              | An unused port                                                | Port for connecting to the builder.                                                                                                          |
+| `prover_port`                   | `Option<u16>`   | `ESPRESSO_PROVER_PORT`               | An unused port                                                | Port for connecting to the prover. If this is not provided, an available port will be selected.                                              |
+| `dev_node_port`                 | `u16`           | `ESPRESSO_DEV_NODE_PORT`             | `20000`                                                       | Port for the dev node. This is used to provide tools and information to facilitate developers debugging.                                     |
+
+## APIs
+
+Once you have successfully run the dev node, you can access the corresponding ports to call the APIs of the
+[`builder`](https://docs.espressosys.com/sequencer/api-reference/builder-api),
+[`sequencer`](https://docs.espressosys.com/sequencer/api-reference/sequencer-api), and `prover`.
+
+In addition, you can access the `dev_node_port` to retrieve debugging information. Here are the details of the dev node
+API.
+
+### GET /api/dev-info
+
+This endpoint returns some debug information for you.
+
+An example response is like this:
+
+```json
+{
+  "builder_url": "http://localhost:41003/",
+  "prover_port": 23156,
+  "l1_url": "http://localhost:8545/",
+  "light_client_address": "0xb075b82c7a23e0994df4793422a1f03dbcf9136f"
+}
+```
+
+### POST /api/set-hotshot-down
+
+This endpoint simulates the effect of a liveness failure of the hotshot consensus protocol in the Light Client smart
+contract.
+
+By calling this, the L1 height in the light contract will be frozen, and rollups will detect the HotShot failure. This
+is intended for testing rollups' functionalities when HotShot is down.
+
+Parameters
+
+| Name   | Type    | Description                              |
+| ------ | ------- | ---------------------------------------- |
+| height | integer | The L1 height from which hotshot is down |
+
+### POST /api/set-hotshot-up
```

Comment:
I'm curious if we really need both endpoints. could we just have `post /api/hotshot` or similar which which reacted on the basis of current state of hotshot (like a toggle). Also since these are POSTs, it might be reasonable to move the command to the POST body. just a thought

Reply:
Yes that makes sense. I don't have any though about which is better. But the parameters of these 2 endpoints are totally different. I think we don't have to combine these into only 1.

Reply:
Humm yea but it is POST, so if we do something like RPC we could just have a command input, and the request bodies would all differ based on the command. But anyway it not really import